packages/loopring_v3/contracts/aux/fast-withdrawal/FastWithdrawalLiquidityProvider.sol

```diff
@@ -81,21 +114,97 @@ contract FastWithdrawalLiquidityProvider is ReentrancyGuard, OwnerManagable
         agent.executeFastWithdrawals{value: value}(withdrawals);
     }
 
-    // Allows the LP to transfer funds back out of this contract.
-    function drain(
-        address to,
+    function deposit(
         address token,
-        uint    amount
+        uint amount
         )
         external
         nonReentrant
-        onlyOwner
     {
+        require(amount > 0, "ZERO_AMOUNT");
+        uint _balance = 0;
         if (token == address(0)) {
-            to.sendETHAndVerify(amount, gasleft()); // ETH
+            _balance = address(this).balance;
+            address(this).sendETHAndVerify(amount, gasleft()); // ETH
         } else {
-            token.safeTransferAndVerify(to, amount);  // ERC20 token
+            _balance = ERC20(token).balanceOf(address(this));
+            token.safeTransferAndVerify(address(this), amount);  // ERC20 token
         }
```

Comment:
Make function `payable` and just check amount received for ETH, use `safeTransferFromAndVerify` for ERC20.